src/Radicle/Ipfs.hs

```diff
@@ -248,11 +251,14 @@ subscribe topic messageHandler = do
         pure ()
   where
     withBackoff :: (MonadLog m, MonadMask m) => m a -> m a
-    withBackoff action
-        = recovering
-            policy
+    withBackoff action = do
+        lastSuccess <- liftIO $ Clock.getTime Clock.Monotonic >>= newIORef
+        recovering
+            (resetAfter lastSuccess (60 * minutes) policy)
             (skipAsyncExceptions ++ [h]) -- `h` should be *after* skipAsync!
-            (const action)
+            . const $
+                action <* liftIO
+                    (Clock.getTime Clock.Monotonic >>= writeIORef lastSuccess)
```

Comment:
`action` never returns in this case (only throws), so it seems like the rhs of `<*` never gets executed? Maybe a catch that rethrows after setting the IORef?

(I'd also maybe rename `lastSuccess` to `lastFailure`)

Reply:
Ah true, brainfart.